README.md

```diff
@@ -1559,28 +1256,357 @@ let main ~config_changed =
+Fibers are scheduled cooperatively within a single domain, but you can also create new domains that run in parallel.
+This is useful to perform CPU-intensive operations quickly.
 
-For example, `Fiber.both f g` always starts running `f` first,
-and only switches to `g` when `f` finishes or performs an effect that can switch fibers.
+There are 2 main ways of achieving this and both are documented below:
+- `Eio.Executor_pool`
+- spawning a one-off domain using `Eio.Domain_manager.run`
 
-Performing IO with external objects (e.g., `stdout`, files, or network sockets) will introduce non-determinism,
-as will using multiple domains.
+The Executor Pool is the recommended way of leveraging OCaml 5's multicore capabilities.
+It's built on top of `Eio.Domain_manager` and offers several advantages:
+- Executor Pool domains (threads) are initialized in advance to minimize overhead. They're also reused, unlike direct usage of `Eio.Domain_manager.run`.
+- Each Executor Pool domain worker can execute multiple jobs concurrently. Whenever a job blocks due to e.g. I/O, the domain's scheduler switches execution to another fiber and/or job.
+- The Executor Pool supports heterogenous jobs efficiently. For example, it will seamlessly run a large number of I/O-bound jobs on one domain while reserving an entire other domain (thread) for a single CPU-bound job. You control this behavior through the `~weight` argument required to submit a job.
 
-Note that `traceln` is unusual. Although it writes (by default) to stderr, it will not switch fibers.
-Instead, if the OS is not ready to receive trace output, the whole domain is paused until it is ready.
-This means that adding `traceln` to deterministic code will not affect its scheduling.
+General multicore notes:
+- `Eio.Std.traceln` can be used safely from multiple domains. It uses a mutex internally, so that trace lines are output atomically.
+- You must ensure that the function passed (to `Eio.Executor_pool.submit` or `Eio.Domain_manager.run`) doesn't have access to any non-threadsafe values. The type system does not check this.
+  - If your multicore code **reads or writes** to/from mutable values, you should read the [Multicore Guide](./doc/multicore.md).
+  - The [Synchronisation Tools](#synchronisation-tools) section documents various threadsafe tools to communicate across domains, such as `Eio.Promise` and `Eio.Stream`.
+  - The [Saturn](https://github.com/ocaml-multicore/saturn) library also offers a collection of efficient threadsafe data structures.
 
-In particular, if you test your code by providing (deterministic) mocks then the tests will be deterministic.
-An easy way to write tests is by having the mocks call `traceln` and then comparing the trace output with the expected output.
-See Eio's own tests for examples, e.g., [tests/switch.md](tests/switch.md).
+### Executor Pool
 
-## Provider Interfaces
+An executor pool distributes jobs among a pool of domains workers (threads).
```

Comment:
I'm not sure we need to mention "thread" here. It can be counter-intuitive. I believe thread here is referring to a thread of execution. In general a thread in OCaml means systhread by default.

Reply:
I thought a lot about this yet I'm still unsure about it, so I appreciate your perspective.

The reasoning was that _I think_ that starting within the next 6 months, Eio will become the single most common entrypoint for OCaml 4 developers to learn about OCaml 5's multicore feature. Within the same timeframe, I also believe that Eio will be **the** place where C#, Java, Go, etc., developers learn how to perform I/O and use multithreading in OCaml. Those developers will be familiar with the concept of threads and I wanted to emphasize that whenever they see the word "domain", they can rest assured that it maps 1:1 to a thread.

Thoughts? What do you think?

Reply:
Fair enough. Maybe add a one-line explanation of what a domain is (and how it relates to the notion of thread in other languages)? We can also link to the manual page on parallelism https://v2.ocaml.org/manual/parallelism.html. I'm still not sure about correlating thread and domain because one domain can host many systhreads.

Reply:
Good idea, done.